Source/Core/Core/DSP/DSPAssembler.cpp

```diff
@@ -71,32 +68,18 @@ bool DSPAssembler::Assemble(const std::string& text, std::vector<u16>& code,
     return false;
 
   // We now have the size of the output buffer
-  if (m_totalSize > 0)
-  {
-    gdg_buffer = (char*)malloc(m_totalSize * sizeof(u16) + 4);
-    if (!gdg_buffer)
-      return false;
-
-    memset(gdg_buffer, 0, m_totalSize * sizeof(u16));
-  }
-  else
+  if (m_totalSize <= 0)
     return false;
 
+  m_output_buffer.resize(m_totalSize);
+
   InitPass(2);
   if (!AssembleFile(file_name, 2))
     return false;
 
-  code.resize(m_totalSize);
-  for (int i = 0; i < m_totalSize; i++)
-  {
-    code[i] = *(u16*)(gdg_buffer + i * 2);
-  }
-
-  if (gdg_buffer)
-  {
-    free(gdg_buffer);
-    gdg_buffer = nullptr;
-  }
+  code = std::move(m_output_buffer);
+  m_output_buffer.clear();
+  m_output_buffer.shrink_to_fit();
```

Comment:
@Orphis In this case I wouldn't necessarily say that not immediately freeing would be a bad thing in the case of assembling multiple files. In the case of assembling only one file, then the assembler instance will probably be discarded shortly after, resulting in memory being freed anyway, but I can switch to just direct assignment if that's clearer/preferable.